packages/blocks/src/api/registration.js

```diff
@@ -758,3 +758,178 @@ export const registerBlockVariation = ( blockName, variation ) => {
 export const unregisterBlockVariation = ( blockName, variationName ) => {
 	dispatch( blocksStore ).removeBlockVariations( blockName, variationName );
 };
+
+/**
+ * Registers a new block bindings source with an object defining its
+ * behavior. Once registered, the source is available to be connected
+ * to the supported block attributes.
+ *
+ * @param {Object}   source                    Properties of the source to be registered.
+ * @param {string}   source.name               The unique and machine-readable name.
+ * @param {string}   source.label              Human-readable label.
+ * @param {Function} [source.getValue]         Function to get the value of the source.
+ * @param {Function} [source.setValue]         Function to update the value of the source.
+ * @param {Function} [source.setValues]        Function to update multiple values connected to the source.
+ * @param {Function} [source.getPlaceholder]   Function to get the placeholder when the value is undefined.
+ * @param {Function} [source.canUserEditValue] Function to determine if the user can edit the value.
+ *
+ * @example
+ * ```js
+ * import { _x } from '@wordpress/i18n';
+ * import { registerBlockBindingsSource } from '@wordpress/blocks'
+ *
+ * registerBlockBindingsSource( {
+ *     name: 'plugin/my-custom-source',
+ *     label: _x( 'My Custom Source', 'block bindings source' ),
+ *     getValue: () => 'Value to place in the block attribute',
+ *     setValue: () => updateMyCustomValue(),
+ *     setValues: () => updateMyCustomValuesInBatch(),
```

Comment:
Have you decided on the final API? There was discussion about deciding about using with voices in favor of `getValues` and `setValues`.

Reply:
I feel confident about using just `getValues` and `setValues`. I was just rebasing the pull request to adapt the registration there: [link](https://github.com/WordPress/gutenberg/pull/63185/files#diff-30ce126b90014d659302087765876898f16562219f796796ed407fdb346c8e4d). My plan is to make the changes needed, leave a comment sharing my opinion, and open it for review.

Reply:
It wasn't very clear what happened to the open discussions that didn't reach any conclusions. When landing changes, it would be helpful for all observers to explain the action plan for next steps in case they would like to continue following the work.

Reply:
I think that's the challenge when documenting the current API that isn't considered a final one.

Reply:
I kept them open to leave time for feedback. I wanted to post a new comment with what I consider the conclusion.

Reply:
I have just posted [this comment](https://github.com/WordPress/gutenberg/pull/63185#issuecomment-2231143154) and marked it as ready for review.